### PR TITLE
vision and rotate to angle works!!!

### PR DESCRIPTION
**Pull request by aaa-png**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/Comc.java
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/Comc.java:1]: What is this and why we have a button for it?

File: src/main/java/edu/greenblitz/robotName/VisionConstants.java [hunk 1]
```diff
@@ -30,9 +30,9 @@ public class VisionConstants {
 	
 	public final static double VISION_TO_STANDARD_DEVIATION = 10;
 	
-	public final static double APRIL_TAG_HEIGHT_METERS = 0.0018;
+	public final static double APRIL_TAG_HEIGHT_METERS = 1.2397;
 	
-	public final static double APRIL_TAG_HEIGHT_TOLERANCE_METERS = 0.0004;
+	public final static double APRIL_TAG_HEIGHT_TOLERANCE_METERS = 0.7;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/VisionConstants.java:35]: Why so high?

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		pidController.setSetpoint(angleSupplier.get().getRadians());
+		pidController.setTolerance(Units.degreesToRadians(10));
+
+
+		double pidOutput = pidController.calculate(
+				SwerveChassis.getInstance().getChassisAngle().getRadians()
+		);
+
+		SmartDashboard.putNumber("setpoint", angleSupplier.get().getDegrees());
+		SmartDashboard.putBoolean("is at setpoint", isAtSetpoint());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java:44]: Remove prints and comments

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java [hunk 3]
```diff
@@ -123,11 +125,17 @@ public void periodic() {
...
 		Logger.processInputs("DriveTrain/Chassis", ChassisInputs);
 		Logger.processInputs("DriveTrain/Gyro", gyroInputs);
-
-		updatePoseEstimatorOdometry();
+		SmartDashboard.putNumber("shoot angle", ShootingStateCalculations.getTargetShooterAngle(ShootingPositionConstants.LEGAL_SHOOTING_ZONE).getDegrees());
+		updatePoseEstimationLimeLight();
 		MultiLimelight.getInstance().recordEstimatedPositions();
 		SmartDashboard.putData(getField());
 	}
+	public void setMaxPower() {
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java:133]: This feels wrong

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java [hunk 4]
```diff
@@ -125,6 +126,8 @@ public void updateStatusSignals(boolean refresh) {
 		}
 	}
 
+	double vel = 0;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java:129]: Debug  var

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java [hunk 5]
```diff
@@ -148,5 +151,10 @@ public void updateInputs(SwerveModuleInputsAutoLogged inputs) {
 		} else {
 			inputs.absoluteEncoderPosition = Units.rotationsToRadians(canCoder.getAbsolutePosition().getValue());
 		}
+
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java:154]: Debug var

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/Comc.java
Comment (bossyshrimp68) [reply src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/Comc.java:1]: we resolved it

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 6]
```diff
@@ -92,14 +101,18 @@ public void initButtons() {
...
-        mainJoystick.L1.whileTrue(new CollectNoteFromFeeder());
-        mainJoystick.Y.onTrue(new InstantCommand(() -> SwerveChassis.getInstance().resetPoseByVision()));
+	public void romyButtons() {
+		mainJoystick.R1.whileTrue(new NoteToShooterForJoystick());
+		mainJoystick.L1.whileTrue(new CollectNoteFromFeeder());
+		mainJoystick.Y.onTrue(new InstantCommand(() -> SwerveChassis.getInstance().resetPoseByVision()));
 
-        SwerveChassis.getInstance().setDefaultCommand(new MoveByJoysticks(ChassisConstants.DRIVE_MODE));
-    }
+		mainJoystick.R1.whileTrue(
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:111]: make a command... pretty sure there is one in shooting service or commands

File: src/main/java/edu/greenblitz/robotName/commands/NoteToShooterForAuto.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+public class NoteToShooterForAuto extends SequentialCommandGroup {
+	
+	public NoteToShooterForAuto(){
+		super(
+				new NoteToIntake(),
+				new NoteFromIntakeToShooter()
+		);
+	}
+	
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/NoteToShooterForAuto.java:16]: delete

File: src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/FlyWheelDefaultCommand.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package edu.greenblitz.robotName.commands.shooter.flyWheel;
+
+import edu.greenblitz.robotName.commands.shooter.funnel.RunFunnelByVelocity;
+
+public class FlyWheelDefaultCommand extends RunFlyWheelByVelocity {
+	public FlyWheelDefaultCommand() {
+		super(50);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/FlyWheelDefaultCommand.java:7]: magic number, and add a space before the func

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+public class RotateToAngleSupplier implements DoubleSupplier {
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java:13]: delete this space, and add spaces between the parameters

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+	public RotateToAngleSupplier(Supplier<Rotation2d> targetAngle) {
+		this(targetAngle, () -> 0);
+	}
+
+	public RotateToAngleSupplier(Supplier<Rotation2d> targetAngle, DoubleSupplier feedForwardSupplier) {
+		this.angleSupplier = targetAngle;
+		this.feedForwardSupplier = feedForwardSupplier;
+		pidController = new PIDController(0.076, 0, 0);
+		pidController.setTolerance(Units.degreesToRadians(2));
+		pidController.enableContinuousInput(-Math.PI,Math.PI);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java:29]: magic numbers

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+	}
+
+	@Override
+	public double getAsDouble() {
+		pidController.setSetpoint(angleSupplier.get().getRadians());
+		pidController.setTolerance(Units.degreesToRadians(10));
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java:37]: magi numer and delete space

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+		double pidOutput = pidController.calculate(
+				SwerveChassis.getInstance().getChassisAngle().getRadians()
+		);
+
+		return pidController.atSetpoint() ? 0 : pidOutput ;
+	}
+
+	public boolean isAtSetpoint() {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngleSupplier.java:46]: why not an isAtAngle? and why here..

File: src/main/java/edu/greenblitz/robotName/subsystems/limelight/Limelight.java [hunk 13]
```diff
@@ -49,11 +48,17 @@ public double getTagHeight() {
 		return poseArray[VisionConstants.getValue(VisionConstants.LIMELIGHT_ARRAY_VALUES.Y_AXIS)];
 	}
 	
+	public boolean getTagConfidence(){
+		boolean tagHeightConfidence = getTagHeight() < VisionConstants.APRIL_TAG_HEIGHT_METERS + VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS || getTagHeight() > VisionConstants.APRIL_TAG_HEIGHT_METERS - VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/limelight/Limelight.java:52]:
```suggestion
		boolean tagHeightConfidence = getTagHeight() < VisionConstants.APRIL_TAG_HEIGHT_METERS + VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS 
		|| getTagHeight() > VisionConstants.APRIL_TAG_HEIGHT_METERS - VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS;
```

File: src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java [hunk 14]
```diff
@@ -68,6 +70,7 @@ public Optional<Pair<Pose2d, Double>> getFirstAvailableTarget() {
 		}
 		return Optional.empty();
 	}
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java:73]: space

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4ISwerveModule.java:1]: sapces

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToSpeaker.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.robotName.commands.swerve;
+
+import edu.greenblitz.robotName.FieldConstants;
+
+public class RotateToSpeaker extends MoveByJoysticks{
+
+	public RotateToSpeaker() {
+		super(
+				DriveMode.NORMAL,
+				() -> new RotateToPoint(() -> FieldConstants.MIDDLE_OF_RED_SPEAKER_POSITION.toTranslation2d()).getAsDouble()
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToSpeaker.java:10]: What happens on blue match?

**Review by yonnbro — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**